src/groupByKey.ts

```diff
@@ -2,6 +2,42 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const workItems: T[] = JSON.parse(JSON.stringify(items));
+  const groupOfObjects: GroupsMap<T> = {};
+
+  workItems.forEach((item) => {
+    const currentKey = item[key] as keyof GroupsMap<T>;
+
+    if (!Object.prototype.hasOwnProperty.call(groupOfObjects, currentKey)) {
+      groupOfObjects[currentKey] = [];
+    }
+
+    groupOfObjects[currentKey].push(item);
+  });
+
+  return groupOfObjects;
 }
+
+// type ForSet = string | number;
+
+// export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+//   const workItems: T[] = JSON.parse(JSON.stringify(items));
+//   const uniqueValues = new Set<ForSet>();
```

Comment:
Passing `string | number` to generic is ok too, but nice that you've moved it to separate type